### PR TITLE
Add ThrowOnError to VideoAnnotationResults

### PR DESCRIPTION
**Pull request by cruzespinal**
I have created a simple test for 'VideoAnnotationResults' in 'Video Intelligence API', I have not created more because the API is still in a 'Beta' state.

Regards!

### CODE REVIEW

**Comment by jskeet** [comment]
There's no point in adding unit tests for `VideoAnnotationResults` - it's generated code, as is everything in this project.

I suggest that if you want to contribute, you discuss ideas of what you might want to contribute with us before you start writing code, otherwise it's very likely you'll just end up wasting your time, I'm afraid. 

I'm not trying to discourage contributions - I'm just saying that as this codebase is somewhat unusual in terms of how much of it is generated, it's worth checking first.

**Comment by jskeet** [comment]
Hang on - I hadn't realized that you'd also added functionality. (That isn't part of the PR description.)

That's *potentially* useful, and we'll need to consider it. I'll edit the PR description to match what's actually in it.

**Comment by jskeet** [comment]
(My advice of checking the code you're proposing before starting to write the code still applies though. It would be very easy for you to write a lot of code and then for us to reject the PR for reasons we could have explained beforehand.)

**Comment by cruzespinal** [comment]
Agree.
ThrowOnError is required in the Container, Dlp, Logging, Debugger and Trace APIs? Do I add them?

Thanks,
John

**Comment by jskeet** [comment]
No, it's not a pattern to be applied universally. If it were, we'd put it in codegen.

It's only useful in cases where a response can be successful at the HTTP level - and potentially still include partial success information - but also represent partial failure.

**Comment by jskeet** [comment]
Okay, I've had another look at this now.

I don't think it's worth adding to just `VideoAnnotationResults`. We could potentially add `ThrowOnAnyError` to `VideoAnnotationResponse`; that would be consistent with the Vision API.

Whether it's worth adding *just* this to the API when there's no other handwritten code is debatable, however - it comes with a maintenance cost in terms of future versions (including API beta versions) where the manual code would need to be propagated. That cost would then be reflected in not being able to spend the time on other work. I believe all the other languages only have auto-generated code for the Video Intelligence API.

@chrisdunelm Thoughts on this? It's reasonably simple code, and I wouldn't hesitate if we already had other handwritten code...

**Comment by chrisdunelm** [comment]
The `ThrowOnError` and potentially `ThrowOnAnyError` (or could still be called `ThrowOnError`)  looks useful, and I think should be included.
I agree about the maintenance cost, but consider it worth it in this instance.

**Comment by jskeet** [comment]
Right - I'll double-check the details of this PR later, merge it, and then add ThrowOnAnyError (name matches the one in Vision) to the response.

File: apis/Google.Cloud.VideoIntelligence.V1/Google.Cloud.VideoIntelligence.V1/VideoAnnotationResultsException.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.VideoIntelligence.V1
+{
+    /// <summary>
+    /// An error occurring when annotating an video.
+    /// </summary>
+    public sealed class VideoAnnotationResultsException : Exception
```
Comment (jskeet) [inline apis/Google.Cloud.VideoIntelligence.V1/Google.Cloud.VideoIntelligence.V1/VideoAnnotationResultsException.cs]: For consistency with Vision, I'll rename this to AnnotateVideoException in a follow-up PR.
(The Vision API has the wrong filename for the class too, which I'll fix.)

**Pull request merged**